src/composition/supergraph/config/resolver/mod.rs

```diff
@@ -0,0 +1,243 @@
+//! This module provides an object that can either produce a [`LazilyResolvedsupergraphConfig`] or a
+//! [`FullyResolvedSupergraphConfig`] and uses the typestate pattern to enforce the order
+//! in which certain steps must happen.
+
+use std::collections::BTreeMap;
```

Comment:
maybe it'd be useful to have that order laid out in an ascii diagram? it might make the relationship between lazy and fully resolved supergraph configs clearer